### PR TITLE
Add reusable functions for routes paths

### LINKED ISSUE #389: Reusable functions for routes paths
Instead of building paths in place everywhere we should have reusable functions with good names / types etc.:

```ts
// shared/path-utils.ts

export const designerPath({projectId, pageId}: {projectId: string, pageId?: string}) => {
  if (pageId) {
    return `/designer/${projectId}?pageId=${pageId}`
  }
  return `/designer/${projectId}`
}

// same for every path that we need in the app...
```

Original discussion: https://github.com/webstudio-is/webstudio-designer/pull/362#discussion_r1013706476

### PR DESCRIPTION
**Pull request by rpominov**
Fixes #389

## Before requesting a review

- [x] if the PR is WIP - use draft mode
- [x] made a self-review
- [x] added inline comments where things may be not obvious (the "why", not "what")
- [x] what kind of review is needed?
  - [x] conceptual
  - [ ] detailed
  - [ ] with testing

## Test cases

- [ ] step by step interaction description and what is expected to happen

## Before merging

- [x] tested locally and on preview environment (preview dev login: 5de6)
- [ ] updated [test cases](https://github.com/webstudio-is/webstudio-designer/blob/main/apps/designer/docs/test-cases.md) document
- [ ] added tests
- [ ] if any new env variables are added, added them to `.env.example` and the `designer/env-check.js` if mandatory

### CODE REVIEW

File: apps/designer/app/shared/router-utils/path-utils.ts [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+
+export const designerPath = ({
+  projectId,
+  pageId,
+}: {
+  projectId: string;
+  pageId?: string;
+}) => `/designer/${projectId}${searchParams({ pageId })}`;
+
+export const dashboardPath = "/dashboard";
```
Comment (kof) [inline apps/designer/app/shared/router-utils/path-utils.ts:25]: should probably be a function for consistency

File: apps/designer/app/shared/router-utils/path-utils.ts [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+export const loginPath = (params: {
+  error?: typeof AUTH_PROVIDERS[keyof typeof AUTH_PROVIDERS];
+  message?: string;
+}) => `/login${searchParams(params)}`;
+
+export const logoutPath = "/logout";
+
+export const googleCallbackPath = "/auth/google/callback";
+
+export const githubCallbackPath = "/auth/github/callback";
```
Comment (kof) [inline apps/designer/app/shared/router-utils/path-utils.ts:36]: could be one function with a provider param

File: apps/designer/app/shared/router-utils/path-utils.ts [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+}: {
+  buildOrigin: string;
+  project: Project;
+  page: Page;
+  mode?: BuildMode;
+}) => {
+  const url = new URL(buildOrigin);
+  url.pathname = page.path;
+
+  if (env.BUILD_REQUIRE_SUBDOMAIN) {
```
Comment (kof) [inline apps/designer/app/shared/router-utils/path-utils.ts:52]: seems better grammar 
```suggestion
  if (env.BUILD_REQUIRES_SUBDOMAIN) {
```

**Review by kof — APPROVED** [review]
looking good

File: apps/designer/app/shared/router-utils/path-utils.ts (hunk 3 above)
Comment (rpominov) [reply apps/designer/app/shared/router-utils/path-utils.ts:52]: A bit of a pain to change at Vercel: need to keep old one for current deployment, then remember to remove it. I'd keep, doesn't seem important

**Pull request merged**